examples/vae_text/vae_train.py

```diff
@@ -154,40 +154,38 @@ def forward(self, data_batch: tx.data.Batch,
         latent_z = dst.rsample()
 
         fc_output = self.mlp_linear_layer(latent_z)
+        labels = text_ids[:, 1:]
         if self._config.decoder_type == "lstm":
             decoder_states = torch.split(
                 fc_output, self.decoder_initial_state_size, dim=1)
 
-            def _embedding_fn(token):
-                embedding = self.decoder_w_embedder(
-                    token)
-                if len(token.size()) > 1:
-                    _latent_z = torch.unsqueeze(latent_z, 1)
-                    _latent_z = _latent_z.repeat([1, embedding.size(1), 1])
-                    embedding = torch.cat([embedding, _latent_z], 2)
-                else:
-                    embedding = torch.cat([embedding, latent_z], 1)
-                return embedding
-
             helper = self.decoder.create_helper(
                 decoding_strategy="train_greedy",
-                embedding=_embedding_fn,
                 start_tokens=start_tokens,
                 end_token=end_token,)
+            latent_z = torch.unsqueeze(latent_z, 0)
+            latent_z = latent_z.repeat([text_ids.size(1), 1, 1])
         else:
             decoder_states = torch.reshape(
                 fc_output, (-1, ) + self.decoder_initial_state_size)
             helper = None
+            text_ids = text_ids[:, :-1]
```

Comment:
Have you tested both RNN & Transformer models? Why is this in the `else` branch?

Reply:
You are right, that can work. Just fix it.